src/views/tutorial-view/index.tsx

```diff
@@ -188,6 +189,18 @@ function TutorialView({
 		},
 	]
 
+	/**
+	 * Keep track of progress for authenticated users
+	 *
+	 * Note that we attach data-ref-id to avoid some
+	 * client-side-navigation-related progress tracking quirks.
+	 */
+	const progressRefsId = `${id}_${collectionCtx.current.id}`
+	const progressRefs = useUpdateTutorialProgress({
```

Comment:
Are you planning on creating a separate hook to return the progress value for this tutorial to render the metadata UI? Or should this hook also return the progress status with the refs?

Reply:
🤔 Perhaps here we could call for the entire collection's progress and just pass this down to the UI (sidebar & metadata). [See example in Learn](https://github.com/hashicorp/learn/blob/master/pages/tutorials/%5B...slug%5D.tsx#L145-L148)

Reply:
The `useUpdateTutorialProgress` hook shouldn't need to return the updated progress value if we leverage React Query like we do for bookmarks. The `useTutorialProgress` hook will automatically return new data if the query key uses is invalidated on successful update. I believe we're already set up correctly for this already because `useTutorialProgress`'s `makeOnMutationSuccess` invokes `setQueryData`.

Reply:
I did a quick test, it works as far as I can tell! So anywhere we use `useTutorialProgress` will automatically get the updated data without any extra work on our part.

Reply:
To confirm Ashlee's comments, plan is:
- This PR makes individual queries. For now this means we fire off a bunch of individual requests, which is not optimal, but gives us convenience from a component architecture standpoint
- In a subsequent PR, I'll add in a batch query at the tutorial view level (similar to [useAllBookmarksById in TutorialCardsGridList, as used on eg collection view](https://github.com/hashicorp/dev-portal/blob/bcae5421fc244c091dc5085d8a1218ace4345dd1/src/components/cards-grid-list/components/tutorial-cards-grid-list/index.tsx#L36).
- The batch query (to be added in next PR) will get fired before the individual queries, and make one request. The subsequent individual queries won't end up making a bunch of requests cause the data they need will already be cached (more optimal! ☺️ )